src/webgl-surface/drawing/shape/curved-line-shape.ts

```diff
@@ -228,4 +251,45 @@ export class CurvedLineShape<T> extends CurvedLine<T> {
     super.setPoints(start, end, controlPoints);
     this.cachedQuadSegments = [];
   }
+
+  containsPoint(point: IPoint) {
+      if (this.type === CurveType.CircularCW || this.type === CurveType.CircularCCW) {
+
+        // Center
+        const center = this.controlPoints[0];
+
+        // Radius
+        const radius = Math.sqrt(Math.pow(this.start.x - center.x, 2) + Math.pow(this.start.y - center.y, 2));
+
+        // Distance from mouse to center
+        const distance = Math.sqrt(Math.pow(point.x - center.x, 2) + Math.pow(point.y - center.y, 2));
+
+        // Angle
+        let angle = getAngle(point, center);
+        const startAngle = getAngle(this.start, center);
+        let endAngle = getAngle(this.end, center);
+
+        if ( startAngle > 0 && endAngle < 0) {
+          if (angle < 0) angle += 2 * Math.PI;
+          endAngle += 2 * Math.PI;
+        }
+
+        // Make sure point is in the endpoint
+        if (
+          distance <= radius + this.lineWidth / 2
+          && distance >= radius - this.lineWidth / 2
```

Comment:
Whenever you are simply doing comparisons of distance, do not calculate the distance with a sqrt, instead compare using the squared values. This way you don't have to use an expensive sqrt operation.